FrontEnd/src/components/basics/ImageUpload/ImageUpload.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { FunctionComponent, ChangeEvent, useState } from 'react';
+import { Row } from 'react-bootstrap';
+import { FilledImage, Button } from '@basics';
+import { imageUpload } from '@icons';
+import styles from './ImageUpload.module.scss';
+import cn from 'classnames';
+
+interface UploadButtonProps {
+  onChange: (e: ChangeEvent<HTMLInputElement>) => void;
+}
+
+const UploadButton: FunctionComponent<UploadButtonProps> = ({
+  onChange,
+}) => (
+  <div className={styles.imageInputWrapper}>
+    <label htmlFor="imageUpload" className={styles.imageInputLabel}>
+      <imageUpload.Add />
+    </label>
+    <input
+      id="imageUpload"
+      type="file"
+      onChange={onChange}
+      multiple
+      accept="image/*"
+      className={styles.imageInput}
+    />
+  </div>
+);
+
+interface IndividualImageProps {
+  index: number;
+  file: File;
+  onDelete: (currentIndex: number) => void;
+}
+
+const IndividualImage: FunctionComponent<IndividualImageProps> = ({
+  index,
+  file,
+  onDelete,
+}) => {
+  const fileUrl = URL.createObjectURL(file);
+
+  return (
+    <div className={styles.singleImage}>
+      <div className={styles.deleteWrapperBackground}>
+        <Button
+          variant="wrapper"
+          className={styles.deleteWrapper}
+          onClick={() => onDelete(index)}
+        >
+          <imageUpload.Delete className={styles.delete} />
+        </Button>
+      </div>
+      <FilledImage
+        src={fileUrl}
+        alt={'uploaded images'}
+        className={styles.roundImage}
+      />
+    </div>
+  );
+};
+
+/* The Beautiful Image Uploader
+Usage: inside the Make a Post when users post images about their housing. Applicable to other features like profile photos as well
+Features: upload up to 6 images and also enable delete upon hover on individual images
+*/
+const ImageUpload: FunctionComponent = () => {
+  const UploadSizeLimit = 6;
+  const [files, setFiles] = useState<File[]>([]);
+  const onChange = (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.target.files && e.target.files.length) {
+      setFiles([...files, ...Array.from(e.target.files)].slice(0, UploadSizeLimit)); // only keep 6 files if user attempt to upload more than 6
```

Comment:
Nitpick: also explain in the comment _why_ the user should be limited to only 6 photos